xmppserver/src/main/java/org/jivesoftware/openfire/update/PluginDownloadManager.java

```diff
@@ -55,16 +56,23 @@ public Update downloadPlugin(String url) {
     /**
      * Installs a new plugin into Openfire.
      *
+     * @param username The user that triggered the install.
      * @param url the url of the plugin to install.
      * @param version the version of the new plugin
      * @param hashCode the matching hashcode of the <code>AvailablePlugin</code>.
      * @return the hashCode.
      */
-    public DownloadStatus installPlugin(String url, String version, int hashCode) {
+    public DownloadStatus installPlugin(String username, String url, String version, int hashCode) {
+
         UpdateManager updateManager = XMPPServer.getInstance().getUpdateManager();
 
         boolean worked = updateManager.downloadPlugin(url);
 
+        if (worked) {
+            final String name = url.substring(url.lastIndexOf('/') + 1);
+            SecurityAuditManager.getInstance().logEvent(username, "downloaded plugin " + name, "Downloaded from: " + url);
+        }
```

Comment:
Shouldn't we be logging an attempt (and the result), rather than only logging if it was successful?

Reply:
Not necessarily I think. This is an _audit_ log, which I think should primarily register events that change the system. A failed attempt to install something doesn't change the system. If we start adding failures, then the audit log quickly becomes a regular log.

Reply:
I've got opinions on this that I'll take to the MUC, but it's not really the topic of this PR. If I convince you, I'll raise a new ticket in Jira 😉